src/middlewares/user.auth.ts

```diff
@@ -1,21 +1,37 @@
 import { Request, Response, NextFunction } from "express";
 import passport from "../config/passport.config";
+import BlacklistedToken from "../models/Blacklist";
 
 const authenticate = (req: Request, res: Response, next: NextFunction) => {
-  passport.authenticate("jwt", { session: false }, (error: any, user: any) => {
-    if (error) {
-      return res.status(500).json({ error: error.message });
-    }
+  passport.authenticate(
+    "jwt",
+    { session: false },
+    async (error: any, user: any) => {
+      const tokenHeader = req.headers.authorization?.split(" ")[1];
+      // console.log("token__________", tokenHeader);
+      if (error) {
+        return res.status(500).json({ error: error.message });
+      }
 
-    if (!user) {
-      return res
-        .status(401)
-        .json({ error: "Unauthorized User. Please login to continue" });
-    }
+      if (!user) {
+        return res
+          .status(401)
+          .json({ error: "Unauthorized User. Please login to continue" });
+      }
```

Comment:
the whole of the below code, needs to be done beforre checking  if (!user) {
So that you send the same response. people do not need to know that the token is blacklisted but just that they are not authentocated

  if (!user || blacklistedToken ) {